src/braket/pennylane_plugin/braket_device.py

```diff
@@ -605,6 +606,61 @@ def execute_and_gradients(self, circuits, **kwargs):
         res = res[0] if len(res) == 1 and active_jac else res
         return res, jacs
 
+    def _is_single_qubit_01_frame(self, f):
+        if self._device.arn == "arn:aws:braket:eu-west-2::device/qpu/oqc/Lucy":
+            return "drive" in f
+        else:
+            raise RuntimeError(
+                f"Single-qubit drive frame for pulse control not defined for "
+                f"device {self._device.arn}"
+            )
+
+    def _is_single_qubit_12_frame(self, f):
+        if self._device.arn == "arn:aws:braket:eu-west-2::device/qpu/oqc/Lucy":
+            return "second_state" in f
+        else:
+            raise RuntimeError(
+                f"Single-qubit drive frame for pulse control not defined for "
+                f"device {self._device.arn}"
+            )
+
+    def _get_frames(self, filter):
+        return {
+            f: info
+            for f, info in self._device.properties.pulse.dict()["frames"].items()
+            if filter(f)
+        }
+
+    @property
+    def settings(self):
+        frames = self._get_frames(filter=self._is_single_qubit_01_frame)
+        frames_12 = self._get_frames(filter=self._is_single_qubit_12_frame)
+
+        drive_frequencies = [frames[f]["frequency"] * 1e-9 for f in frames]  # Hz to GHz
+        device_info = self._device.properties.paradigm.dict()
+
+        connections = []
+
+        for q1, connected_qubits in device_info["connectivity"]["connectivityGraph"].items():
+            for q2 in connected_qubits:
```

Comment:
Nice! This should also work for the Rigetti device already :)

Reply:
There is one issue with the anharmonicity, as one qubit on the Rigetti device doesn't have a second transition frequency, so just zipping them together doesn't work. Also, we would have to decide if we want sequential wire labelling (as implemented) or if we would want the wire number to match the qubit number on the Rigetti system. But otherwise everything should be transferrable to Rigetti, its just a question of adding a filter to the filter functions! 

So just a few tweaks to make it compatible with both.